dakara_server/library/management/commands/createworks.py

```diff
@@ -0,0 +1,258 @@
+import os
+import sys
+import logging
+import importlib
+
+from django.core.management.base import BaseCommand, CommandError
+
+from library.models import (WorkType, WorkAlternativeTitle, Work)
+
+from .feed_components import default_work_parser
+
+# get logger
+logger = logging.getLogger(__name__)
+
+
+class WorkCreator:
+    """Work creator and updater. Create and update works in the
+    database provided a work file and a parser.
+
+    Args:
+        work_file (str): Absolute path of the file storing the works data.
+        parser (module): Custom python module used to extract data from file.
+        dry_run (bool): Run script in test mode.
+
+    About parser:
+        This module should define a method called `parse_work` which takes
+        a file path as argument and return a dictionnary with the following:
+            works (dict): keys are a query name worktype, values are
+            a list such that each element of this list is a dictionnary with
+            the following keys:
+                    title (str): title of the work
+                    subtitle (str): subtitle of the work
+                    alternative_titles (list): list of alternative names of
+                    a work
+    """
+    def __init__(
+            self,
+            work_file,
+            parser,
+            debug):
+        self.work_file = work_file
+        self.parser = parser
+        self.debug = debug
+
+    def remove_incorrect_works(self, work_listing):
+        """Remove works that have an incorrect structure in a list of works.
+
+        Return
+            A new list with the incorrect works removed."""
+
+        work_to_remove = []
+
+        logger.debug("Removing the incorrect work from the current work type.")
+        for dict_work in work_listing:
+
+            # check if it is a dictionary
+            if not isinstance(dict_work, dict):
+                work_to_remove.append(dict_work)
+                logger.debug("Incorrect work : value is not a dictionary.")
```

Comment:
You should give a clue on which work has an incorrect structure, for debugging purpose. The best way is to give the list entry number:
```python
for index, dict_work in enumerate(work_listing):
    ...
    logger.debug("Incorrect work at index {}: value must be a dictionary".format(index))
```